contracts/implementations/streamr/VehicleStream.sol

```diff
@@ -0,0 +1,122 @@
+        address indexed subscriber,
+        uint256 subscribeExpiration
+    );
+
+    /**
+     * @notice Creates a vehicle stream associated with a vehicle id
+     * @dev Stream is is in format streams.dimo.eth/vehicle/<vehicleId>
+     * @dev Reverts if vehicle id does not exist or caller is not the owner
+     * @param vehicleId Vehicle node Id
+     */
+    function createVehicleStream(uint256 vehicleId) external {
+        address vehicleIdProxyAddress = VehicleStorage
+            .getStorage()
+            .idProxyAddress;
+        IStreamRegistry streamRegistry = IStreamRegistry(
+            StreamrManagerStorage.getStorage().streamRegistry
+        );
+
+        try INFT(vehicleIdProxyAddress).ownerOf(vehicleId) returns (
+            address vehicleOwner
+        ) {
+            if (vehicleOwner != msg.sender) {
+                revert Errors.Unauthorized(msg.sender);
+            }
+        } catch {
+            revert Errors.InvalidNode(vehicleIdProxyAddress, vehicleId);
+        }
+
+        string memory streamPath = string(
+            abi.encodePacked("/vehicle/", Strings.toString(vehicleId))
+        );
+        streamRegistry.createStreamWithENS(DIMO_STREAM_ENS, streamPath, "{}");
+
+        string memory streamId = string(
+            abi.encodePacked(
+                DIMO_STREAM_ENS_VEHICLE,
+                Strings.toString(vehicleId)
+            )
+        );
+
+        emit VehicleStreamAssociated(streamId, vehicleId);
+    }
+
+    /**
+     * @notice Set a subcription permission for a user
+     * @dev Only the vehicle id owner can add new subscribers to their stream
+     * @param vehicleId Vehicle node id
+     * @param subscriber Vehicle stream subscriber
+     * @param subscribeExpiration Subscription expiraton timestamp
+     */
+    function subscribeToVehicleStream(
+        uint256 vehicleId,
+        address subscriber,
+        uint256 subscribeExpiration
+    ) external {
+        address vehicleIdProxyAddress = VehicleStorage
+            .getStorage()
+            .idProxyAddress;
+        StreamrManagerStorage.Storage storage sms = StreamrManagerStorage
+            .getStorage();
+        IStreamRegistry streamRegistry = IStreamRegistry(sms.streamRegistry);
+
+        // TODO To be possibly replaced by signature verification (like permit)
+        try INFT(vehicleIdProxyAddress).ownerOf(vehicleId) returns (
+            address vehicleOwner
+        ) {
+            if (vehicleOwner != msg.sender) {
+                revert Errors.Unauthorized(msg.sender);
+            }
+        } catch {
+            revert Errors.InvalidNode(vehicleIdProxyAddress, vehicleId);
+        }
+
+        string memory streamId = string(
+            abi.encodePacked(
+                DIMO_STREAM_ENS_VEHICLE,
+                Strings.toString(vehicleId)
+            )
+        );
+
+        // Creates the subscription permission setting subscribeExpiration
+        streamRegistry.setPermissionsForUser(
```

Comment:
I worry that we may be clobbering some permissions that the `subscriber` already does have. What if we call [`setExpirationTime`](https://github.com/streamr-dev/network-contracts/blob/08dfadac4999d30f1e84baae2d4ba807843f5d5b/packages/network-contracts/contracts/StreamRegistry/StreamRegistryV4.sol#L311)?

Reply:
For example, someone might accidentally (or not) remove the DIMO Streamr node's rights to a stream with this function!

Reply:
Makes sense. I didn't want to make two external calls, but looks necessary here. We might want to create a more generic permissions setting in the future.

Reply:
About the two calls. Looks like is not necessary call `grantPermission` first, since `setExpirationTime` would change the state we need right away. However, it does not emit `PermissionUpdated` by its own and Juuso showed me with the two call at first.

Reply:
Oh, I didn't realize it was two calls. Hum. I thought the timestamp was everything.

Reply:
So it's going to set it and generate an event saying the subscribe expiration is `MAX_INT`, and then set it to... this seems like a bug on their end.